python/ml4ir/base/model/architectures/auto_dag_network.py

```diff
@@ -81,12 +84,8 @@ def __init__(
         self.inputs_as_list = inputs_as_list
 
     def __str__(self):
-        """Node is identified by its name"""
-        return self.name
-
-    def __repr__(self):
-        """Get the obj representation"""
-        return self.name
+        """Get a readable representation of the node"""
+        return f"{self.name}" if self.is_input_node else f"{self.name}\n[{self.layer.__class__.__qualname__}]"
```

Comment:
@arvindsrikantan Could add input node names too.

Reply:
That makes the visualization very clunky.